### PR TITLE
fix: workaround decode error caused by large key

### PR DESCRIPTION
**Pull request by hzxa21**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

This is a hotfix to workaround decode error caused by store key larger than 64KB. In the future, we may come up with a better way to prevent large key from occuring in the storage (e.g. limit the state table primary key size).

Note that this change breaks the format compatibility but in a very limited manner (only if there are keys with exact len=64KB written prior to this PR).

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
~~- [] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features).~~
~~- [] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)~~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/storage/src/hummock/sstable/block.rs [hunk 1]
```diff
@@ -147,16 +147,26 @@ pub struct KeyPrefix {
     offset: usize,
 }
 
+const MAX_KEY_LEN: usize = u16::MAX as usize;
+
 impl KeyPrefix {
     pub fn encode(&self, buf: &mut impl BufMut) {
         buf.put_u16(self.overlap as u16);
-        buf.put_u16(self.diff as u16);
+        if self.diff >= MAX_KEY_LEN {
```
Comment (soundOfDestiny) [inline src/storage/src/hummock/sstable/block.rs:155]: comments in L252 needs to be fixed

**Review by soundOfDestiny — APPROVED** [review]
rest LGTM

File: src/storage/src/hummock/sstable/block.rs [hunk 2]
```diff
@@ -147,16 +147,26 @@ pub struct KeyPrefix {
     offset: usize,
 }
 
+const MAX_KEY_LEN: usize = u16::MAX as usize;
+
 impl KeyPrefix {
     pub fn encode(&self, buf: &mut impl BufMut) {
         buf.put_u16(self.overlap as u16);
```
Comment (soundOfDestiny) [inline src/storage/src/hummock/sstable/block.rs:154]: we should change our algorithm to make sure `self.overlap<=65536`

**Review by soundOfDestiny — APPROVED** [review]
rest LGTM

**Review by wcy-fdu — APPROVED** [review]

**Review by wenym1 — APPROVED** [review]
Rest LGTM

File: src/storage/hummock_sdk/src/key.rs [hunk 3]
```diff
@@ -25,6 +25,8 @@ use crate::HummockEpoch;
 
 pub const EPOCH_LEN: usize = std::mem::size_of::<HummockEpoch>();
 pub const TABLE_PREFIX_LEN: usize = std::mem::size_of::<u32>();
+// Max length for key overlap and diff length. See KeyPrefix::encode.
+pub const MAX_KEY_LEN: usize = u16::MAX as usize;
```
Comment (Li0k) [inline src/storage/hummock_sdk/src/key.rs:29]: ```entry (kv pair): | overlap len (2B) | diff len (2B) | value len(4B) | diff key | value |```

Would it be more intuitive to add format to the comment?

**Review by Li0k — APPROVED** [review]
Rest LGTM, thanks for the PR

**Pull request merged**